src/main/java/gov/cabinetofice/gapuserservice/web/UserController.java

```diff
@@ -46,10 +47,15 @@ public ResponseEntity<UserDto> getUserById(HttpServletRequest httpRequest, @Path
     @PatchMapping("/user/{userId}/department")
     public ResponseEntity<User> updateDepartment(HttpServletRequest httpRequest, @PathVariable("userId") Integer userId,
                                                    @RequestParam(value = "departmentId", required = false) Integer departmentId) {
+
         if (!roleService.isSuperAdmin(httpRequest)) {
             throw new ForbiddenException();
         }
 
+        if(oneLoginUserService.isUserApplicantAndFindOnly(oneLoginUserService.getUserById(userId))) {
+            throw new InvalidRequestException("User only has find and applicant roles");
```

Comment:
maybe a message like "Users with find and applicant roles cannot be assigned a department"